src/useFind.ts

```diff
@@ -112,10 +124,42 @@ export default <CustomApplication extends Application>(feathers: CustomApplicati
       }
 
       try {
+        const originalParams: Params = params.value;
+        const originalQuery: Query & { $limit?: number } = originalParams.query || {};
         // TODO: the typecast below is necessary due to the prerelease state of feathers v5. The problem there is
         // that the AdapterService interface is not yet updated and is not compatible with the ServiceMethods interface.
-        const res = await (service as unknown as ServiceMethods<M>).find(params.value);
-        data.value = Array.isArray(res) ? res : [res];
+        const res = await (service as unknown as ServiceMethods<M> | AdapterService<M>).find(originalParams);
+        if (call !== currentFindCall.value) {
+          // stop handling response since there already is a new find call running within this composition
+          return;
+        }
+        if (isPaginated(res)) {
```

Comment:
What about using three if blocks with returns here: `isPaginated(res) && useAllPages`, `isPaginated(res) && !useAllPages` and `!isPaginated(res)`. That could make it easier to read.

Reply:
@anbraten Good idea. I will do that

Reply:
@anbraten I made the order of the cases so that the most common use case is handled first

Reply:
perfect.